lib/src/ui/pages/home/home_page.dart

```diff
@@ -116,6 +120,39 @@ class _HomePageState extends State<HomePage> {
     );
   }
 
+  Widget _buildClickableTitle(BuildContext context) {
+    return DefaultTextStyle.merge(
+      style: Theme.of(context).textTheme.headline2.copyWith(
+            fontFamily: _Constants.appBarTitleFontFamily,
+          ),
+      child: GestureDetector(
+        onTap: () {
+          setState(() {
+            _articleFuture = _nyTimesRepository.getFirstMostPopularArticle();
+          });
+        },
+        child: _articleFuture == null
+            ? Text(UntranslatableStrings.newYorkTimes)
+            : FutureBuilder<Article>(
+                future: _articleFuture,
+                builder: (context, snapshot) {
+                  if (snapshot.connectionState == ConnectionState.waiting) {
+                    return CircularProgressIndicator();
+                  } else if (snapshot.hasError) {
+                    var exception = ApiException.fromDioError(snapshot.error);
+                    return Text(ErrorMessageProvider.provideErrorMessage(exception, context));
```

Comment:
przechwycenie błędu docelowo będzie w klasie viewmodelu/blocu zależnie od tego jaką architekturę wybierzemy

Reply:
jeżeli zostaniemy przy ChangeNotifierach to tam też on nam zwróci stan